edisgo/flex_opt/reinforce_measures.py

```diff
@@ -725,3 +732,379 @@ def _replace_by_parallel_standard_lines(lines):
+                    "mv_lv_transformer"
+                ]
+            ]
+        except KeyError:
+            raise KeyError("Standard MV/LV transformer is not in the equipment list.")
+
+        transformers_changes = {"added": {}}
+
+        transformer_s = grid.transformers_df.iloc[0]
+        new_transformer_name = transformer_s.name.split("_")
+        grid_id_ind = new_transformer_name.index(str(grid.id))
+        new_transformer_name[grid_id_ind] = f"{str(grid.id)}1001"
+
+        transformer_s.s_nom = standard_transformer.S_nom
+        transformer_s.type_info = None
+        transformer_s.r_pu = standard_transformer.r_pu
+        transformer_s.x_pu = standard_transformer.x_pu
+        transformer_s.name = "_".join([str(_) for _ in new_transformer_name])
+        transformer_s.bus0 = bus_mv
+        transformer_s.bus1 = bus_lv
+
+        new_transformer_df = transformer_s.to_frame().T
+
+        edisgo_obj.topology.transformers_df = pd.concat(
+            [edisgo_obj.topology.transformers_df, new_transformer_df]
+        )
+        transformers_changes["added"][
+            f"LVGrid_{str(grid.id)}1001"
+        ] = new_transformer_df.index.tolist()
+
+        return transformers_changes
+
+    G = grid.graph
+    station_node = list(G.nodes)[0]  # main station
+
+    relevant_lines = grid.lines_df.loc[
+        (grid.lines_df.bus0 == station_node) | (grid.lines_df.bus1 == station_node)
+    ]
+
+    logger.debug(f"{grid} has {len(relevant_lines)} feeder.")
+
+    paths = {}
+    first_nodes_feeders = {}
+
+    for node in G.nodes:
+        path = nx.shortest_path(G, station_node, node)
+
+        for first_node in relevant_lines.bus1.values:
+            if first_node in path:
+                paths[node] = path
+                first_nodes_feeders.setdefault(path[1], []).append(
+                    node  # first nodes and paths
+                )
+
+    lines_changes = {}
+    transformers_changes = {}
+    nodes_tb_relocated = {}  # nodes to be moved into the new grid
+
+    # note: The number of critical lines in the Lv grid can be more than 2. However,
+    # if the node_1_2 of the first feeder in the for loop is not the first node of the
+    # feeder, it will add data frames even though the following feeders only 1 node
+    # (node_1_2=first node of feeder). In this type of case,the number of critical lines
+    # should be evaluated for the feeders whose node_1_2 s are not the first node of the
+    # feeder. The first check should be done on the feeders that have fewer nodes.
+
+    first_nodes_feeders = dict(
+        sorted(
+            first_nodes_feeders.items(), key=lambda item: len(item[1]), reverse=False
+        )
+    )
+
+    count_inept = 0
+
+    for first_node, nodes_feeder in first_nodes_feeders.items():
+        # first line of the feeder
+        first_line = relevant_lines[relevant_lines.bus1 == first_node].index[0]
+
+        # the last node of the feeder
+        last_node = nodes_feeder[-1]
```

Comment:
I am not sure if that is correct. nodes_feeder contains the buses in the feeder in random order, doesn't it?

Reply:
As `nx.shortest_path` 

https://github.com/openego/eDisGo/blob/ac17817920b458a6bcef201dc54f3afee30c0762/edisgo/flex_opt/reinforce_measures.py#L945

returns the nodes in order from first to last I replaced the nodes in the `first_nodes_feeders` values with the sorted nodes of the main feeder determined by number of nodes in `paths`:

https://github.com/openego/eDisGo/blob/ac17817920b458a6bcef201dc54f3afee30c0762/edisgo/flex_opt/reinforce_measures.py#L969-L976

This way `node_1_2` i guaranteed to be part of the main path as well. In combination with `get_downstream_buses`

https://github.com/openego/eDisGo/blob/ac17817920b458a6bcef201dc54f3afee30c0762/edisgo/flex_opt/reinforce_measures.py#L1032

it is ensured that all nodes of branched of parts of the feeder are still relocated to the new grid.